tsdb/block.go

```diff
@@ -568,6 +568,9 @@ func (pb *Block) CleanTombstones(dest string, c Compactor) (*ulid.ULID, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	pb.Close()
```

Comment:
We should not be closing the block here, we should let the DB reload do the cleanup.

Reply:
That was just an experimental thing on the last commit... I will just revert the last commit, but the tests were not passing before as well.